drivers/prior_living_situation/app/models/prior_living_situation/details.rb

```diff
@@ -0,0 +1,136 @@
+###
+# Copyright 2016 - 2021 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module
+  PriorLivingSituation::Details
+  extend ActiveSupport::Concern
+  included do
+    def detail_link_base
+      "#{section_subpath}details"
+    end
+
+    def section_subpath
+      "#{self.class.url}/"
+    end
+
+    def detail_path_array
+      [:details] + report_path_array
+    end
+
+    def detail_hash
+      {}.merge(living_situation_detail_hash)
+    end
+
+    def living_situation_detail_hash
+      @living_situation_detail_hash ||= {}.tap do |hashes|
+        data_for_living_situations.each do |population, data|
+          if population == :all
+            data.each do |title, client_ids|
+              key = [population, title].join('_').underscore
```

Comment:
If `title` includes spaces, they will end up in the key -- I believe this is fine, but since you are joining w/ `_` I want to make sure.

Reply:
Should be ok, as long as they are consistent with the view side, which as far as I could tell is the case.